src/color.rs

```diff
@@ -180,7 +223,23 @@ impl<T: Primitive + 'static> Pixel for $ident<T> {
 
     const COLOR_MODEL: &'static str = $interpretation;
 
-    const COLOR_TYPE: ColorType = $color_type;
+    const CHANNELS_TYPE: ChannelsType = $channels_type;
+
+    fn color_type() -> ColorType {
+        match (Self::CHANNELS_TYPE, std::mem::size_of::<T>()) {
+            (ChannelsType::L, 1) => ColorType::L8,
+            (ChannelsType::L, _) => ColorType::L16,
+            (ChannelsType::La, 1) => ColorType::La8,
+            (ChannelsType::La, _) => ColorType::La16,
+            (ChannelsType::Rgb, 1) => ColorType::Rgb8,
+            (ChannelsType::Rgb, _) => ColorType::Rgb16,
+            (ChannelsType::Rgba, 1) => ColorType::Rgba8,
+            (ChannelsType::Rgba, _) => ColorType::Rgba16,
+            (ChannelsType::Bgr, _) => ColorType::Bgr8,
+            (ChannelsType::Bgra, _) => ColorType::Bgra8,
+            (_, _) => unreachable!(),
```

Comment:
I don't love matching on the size of the subpixel type, but I'm not sure of a better option for now either. Though maybe we should panic any time the size isn't 1 or 2 bytes?

Reply:
Prior to this PR if one was using another primitive type like `f32`, `color_type` wouldn't panic, so that's the behavior I preserved. This does change the behavior because one would now get `ColorType::Rgb16` for an `Rgb<f32>` type instead of `ColorType::Rgb8`, but that seems appropriate or at least less wrong. If `ChannelsType` remains, I could see an argument for having this panic or behave differently for any non-`u8`/`u16` type. However, if `ChannelsType` is removed and this panics, there's no way to do channel-based logic indifferent to the subpixel type for non-`u8`/`u16` subpixel types (e.g., give me the red channel index for this `Pixel`).

Reply:
It should match explicitely on `ChannelsType::__NonExhaustive` though, to make it clear that this is actually unreachable.